### ISSUE TITLE
change the IDs if devices

### ISSUE THREAD

**Issue by My1** [evilC/UCR#170]
in a multi-controller merging setup it may happen that you connect the controllers in the wrong order or something else and the IDs are completely messed up. it would be nice if there would be a feature that allows to change all Stick-IDs assigned in a profile to new devices, in a fashion like this:
click "switch devices" or something like that in the menu
it will ask for each stick-ID that is assigned in the profile for a button press on the respective controller, and change all of the addons connected to that old ID to the new one.

**Comment by evilC** [comment]
You can use [PJP JoyIDs](http://theairtacticalassaultgroup.com/forum/showthread.php?t=13009) to change the IDs of your devices.  
The order in which you plug in devices for any one given boot should not alter the ID.  

Furthermore, Joystick "ID" (ie index number) is a concept of the WinMM API, which AHK uses to read joysticks.  
This is fundamentally flawed, as it only allows reading from 6 axes, 32 buttons, 1 POV, whereas windows supports 8 axes, 128 buttons, 4 POVs.  

Just last weekend I wrote a [POC for replacing AHK's joystick support with a new DirectX-based system](https://autohotkey.com/boards/viewtopic.php?f=19&t=28889).  
Once I implement this into UCR, joystiicks will be keyed to GUID, which never changes.  

This system will also mean that I can present joystick names instead of ID numbers, which should be a huge improvement.

**Comment by My1** [comment]
oh nice Idea. although it might then be helpful to replace device in the profile anyways e.g. if you wanna exchange profiles with others.

**Comment by evilC** [comment]
Yes, this is the one downside of using GUIDs or names over "Stick 1" - you lose portability.  
Good suggestion though, if I implement this feature, I will look into providing something to replace all instances of one GUID with another.
